tests/testthat/test-lancelot-vaccination.R

```diff
@@ -3220,3 +3220,222 @@ test_that("run sensible vaccination schedule with waning and boosters", {
+  info <- mod$info()
+  y0 <- lancelot_initial(info, 1, p)$state
+  mod$update_state(state = lancelot_initial(info, 1, p)$state)
+  y <- mod$transform_variables(drop(mod$simulate(seq(0, 400, by = 4))))
+
+  expect_true(all(y$N_tot - mod$transform_variables(y0)$N_tot == 0))
+  expect_true(all(y$N_tot_sero_1 -
+                    mod$transform_variables(y0)$N_tot_sero_1 == 0))
+  expect_true(all(y$N_tot_sero_2 -
+                    mod$transform_variables(y0)$N_tot_sero_2 == 0))
+  expect_true(all(y$N_tot_PCR - mod$transform_variables(y0)$N_tot_PCR == 0))
+  expect_true(all(colSums(y$N_tot) - y$N_tot_sero_1 == 0))
+  expect_true(all(colSums(y$N_tot) - y$N_tot_sero_2 == 0))
+  expect_true(all(colSums(y$N_tot) - y$N_tot_PCR == 0))
+})
+
+
+test_that("No vaccine skip moves when expected", {
+  vaccine_schedule <- test_vaccine_schedule(booster_daily_doses = 10000)
+
+  p <- lancelot_parameters(1, "london",
+                           rel_susceptibility = rep(1, 5),
+                           vaccine_index_dose2 = 2L,
+                           vaccine_index_booster = 4L,
+                           vaccine_schedule = vaccine_schedule,
+                           vaccine_progression_rate = c(0, 0, 1 / 10, 0, 0),
+                           n_doses = 3,
+                           vacc_skip_from = 3L,
+                           vacc_skip_to = 5L,
+                           vacc_skip_weight = 0,
+                           vacc_skip_progression_rate = 0)
+  np <- 3L
+  mod <- lancelot$new(p, 0, np, seed = 1L)
+  info <- mod$info()
+
+  state <- lancelot_initial(info, 1, p)$state
+
+  mod$update_state(state = state)
+
+  set.seed(1)
+  y <- mod$simulate((1:220) * 4)
+  y <- mod$transform_variables(drop(y))
+
+  ## no vaccine skip moves (boosting before waning)
+  expect_vector_equal(y$cum_n_vacc_skip[, , 220], 0)
+  ## but lots of people getting third doses
+  expect_true(sum(y$cum_n_vaccinated[, 4, , 220]) > 1e6)
+})
+
+
+test_that("Everyone vaccine skips if there is no waning", {
+  vaccine_schedule <- test_vaccine_schedule(booster_daily_doses = 10000)
+
+  p <- lancelot_parameters(1, "london",
+                           rel_susceptibility = rep(1, 5),
+                           vaccine_index_dose2 = 2L,
+                           vaccine_index_booster = 4L,
+                           vaccine_schedule = vaccine_schedule,
+                           vaccine_progression_rate = c(0, 0, 0, 0, 0),
+                           n_doses = 3,
+                           vacc_skip_from = 3L,
+                           vacc_skip_to = 5L,
+                           vacc_skip_weight = 1,
+                           vacc_skip_progression_rate = 0)
+  np <- 3L
+  mod <- lancelot$new(p, 0, np, seed = 1L)
+  info <- mod$info()
+
+  state <- lancelot_initial(info, 1, p)$state
+
+  mod$update_state(state = state)
+
+  set.seed(1)
+  y <- mod$simulate((1:200) * 4)
+  y <- mod$transform_variables(drop(y))
+
+  ## All booster moves are via vaccine skip
```

Comment:
could also check here that this is not zero i.e. people are indeed receiving boosters

Reply:
Done